<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1099</width>
    <height>945</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>200</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Maze&amp;Cave</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>620</y>
      <width>181</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Размеры пещеры</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="born_limit">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>750</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>7</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="cave_initialize_chance">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>720</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>959</x>
      <y>730</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Лимит смерти</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generate_cave">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>570</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Сгенерировать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_cave">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>570</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить пещеру</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="death_limit">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>750</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>7</number>
    </property>
   </widget>
   <widget class="QPushButton" name="start_updating_cave">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>830</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Запустить</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>770</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Время оновления (мс)</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="cave_rows">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>650</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="QPushButton" name="update_cave">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>650</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Обновить</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stop_updating_cave">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>880</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Остановить</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>819</x>
      <y>730</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Лимит рождения</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="cave_columns">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>650</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="CaveWidget" name="Cavewidget" native="true">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>20</y>
      <width>501</width>
      <height>491</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="cave_update_time">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>800</y>
      <width>201</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>1000000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>690</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Шанс рождения</string>
    </property>
   </widget>
   <widget class="GLWidget" name="GLwidget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>501</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="maze_rows">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>590</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QPushButton" name="save_maze_to_file">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>620</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить лабиринт</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>750</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X второй ячейки</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="i1">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>670</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>49</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="i2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>750</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>49</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="load_maze">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>720</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить лабиринт</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>700</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y первой ячейки</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="maze_columns">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>590</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="j1">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>700</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>49</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>670</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X первой ячейки</string>
    </property>
   </widget>
   <widget class="QPushButton" name="solve_maze">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>820</y>
      <width>231</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Найти путь</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>780</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y второй ячейки </string>
    </property>
   </widget>
   <widget class="QPushButton" name="generate_maze">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>620</y>
      <width>211</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Сгенерировать</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>560</y>
      <width>211</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Размер лабиринта</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="j2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>780</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>49</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>view/glwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CaveWidget</class>
   <extends>QWidget</extends>
   <header>view/cavewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
